packages/react/src/reconciler.ts

```diff
@@ -7,6 +7,12 @@ import type {
   RemoteComponentType,
 } from '@remote-ui/core';
 
+const FUNCTION_CURRENT_IMPLEMENTATION_KEY = '__current';
+
+type HotSwappableFunction<T extends Function> = T & {
+  [FUNCTION_CURRENT_IMPLEMENTATION_KEY]: any;
```

Comment:
Can we be more strict than `any` since we know it's 
```
{
    enumerable: false,
    configurable: false,
    writable: true,
    value: T,
}
```

Reply:
That's the property descriptor; it means that it does definitely start out as `{[FUNCTION_CURRENT_IMPLEMENTATION_KEY]: T}`, but it does not necessarily stay that way. After this is sent over the wire, the original function might change to an `undefined`, in which case `[FUNCTION_CURRENT_IMPLEMENTATION_KEY]: undefined`. That's why I widened the type on this property.